feat: Promote random tryte generation as generic utility

splasky opened the pull request:
Defining gen_tryte() inside main.c may cause us hard to integrate with legato or implement unit-test. A new file tryte_msg.h is added, it includes for tryte message related implementation.

Add gen_trytres_message() for generate unify trytes message for logger and https sender.

Closes #50

howjmay commented on utils/tryte_msg.h:
```diff
@@ -0,0 +1,31 @@
+/**
+ * @brief generate trytes from random
+ *
+ * @param len length of output
```
@param[in] len length of output

howjmay commented on utils/tryte_msg.h:
```diff
@@ -0,0 +1,31 @@
+ * @brief generate trytes from random
+ *
+ * @param len length of output
+ * @param out output array
```
@param[out] out output array

howjmay commented on utils/tryte_msg.h:
```diff
@@ -0,0 +1,31 @@
+/**
+ * @brief create message to send to tangle accerlator
+ *
+ * @param tryte_msg byte message which has convert to tryte
```
ditto

howjmay commented on utils/tryte_msg.c:
```diff
@@ -0,0 +1,26 @@
+  }
+}
+
+void gen_tryte_msg(const char *tryte_msg, const uint8_t *addr, char req_body[1024]) {
```
Why do we need this function? This function is really a general function, and can be expanded in the place it is called.

splasky replied:
`get_tryte_msg` is create for two reason:
1. Hidden request body from main function. Putting request body inside main function will cause difficulties for port to new device and hard to reuse.
2. Making unified trytes message will let logger and `send_http_msg()` to log and send message with same format.

howjmay commented:
The way we send message to tangle-accelerator is still under discussion.
See issue [#518](https://github.com/DLTcollab/tangle-accelerator/issues/518) for more information

`gen_trytres_message()` may be deprecated at that time.

howjmay reviewed (changes requested):
We use rebase instead of merge

jserv commented on utils/tryte_msg.c:
```diff
@@ -0,0 +1,26 @@
+  snprintf(req_body, 1024, REQ_BODY, tryte_msg, addr);
+
+#ifdef DEBUG
+  printf("req packet = \n%s", req_body);
```
It is weird to place debug messages here since there is no explicit req packets.

jserv replied:
I felt the bad smell regarding `gen_tryte_msg` which violates the naming scheme.

jserv reviewed (changes requested):
Change the subject to reflect pull request.

jserv commented on utils/tryte_msg.c:
```diff
@@ -0,0 +1,22 @@
+
+void gen_rand_trytes(uint16_t len, uint8_t *out) {
+  const char tryte_alphabet[] = "9ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  uint8_t rand_index;
```
Reduce the scope. Move the declaration to Line 14.

jserv commented on utils/tryte_msg.c:
```diff
@@ -0,0 +1,22 @@
+  const char tryte_alphabet[] = "9ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  uint8_t rand_index;
+  for (int i = 0; i < len; i++) {
+    rand_index = rand() % 27;
```
Avoid using hardcoded `27`. Instead, use `sizeof(tryte_alphabet)/size(char)`.

jserv reviewed (changes requested):
The filename `utils/tryte_msg` might be misleading. We can merge these function to more generic one, `utils/trytes.h`.

jserv commented on utils/trytes.h:
```diff
@@ -0,0 +1,31 @@
+/**
+ * @brief create unify message to send to tangle accerlator
+ *
+ * @param[in] tryte_msg byte message which has convert to tryte
```
Fix grammar.

jserv commented on utils/trytes.h:
```diff
@@ -0,0 +1,31 @@
+ * @brief generate trytes from random
+ *
+ * @param[in] len length of output
+ * @param[out] out output array
```
Should be "the pointer to output array".

jserv commented on utils/trytes.c:
```diff
@@ -0,0 +1,30 @@
+#include <stdlib.h>
+#include <string.h>
+
+#define REQ_BODY                                                           \
```
I am not sure if it makes to duplicate the request body template here.
Defer to @HowJMay

splasky replied:
Mark new issue #60 for this.

Commit:
feat: Promote random tryte generation as generic utility

Defining gen_tryte() inside main.c may cause us hard to integrate
with legato or implement unit-test. A new file tryte_msg.h is added,
it includes for tryte message related implementation.

Add gen_trytres_message() for generate unify trytes message for logger and https sender.

Closes #50

jserv commented on utils/trytes.h:
```diff
@@ -0,0 +1,31 @@
+#include <stdint.h>
+
+/**
+ * @brief Generate trytes from random
```
Should be randomized trytes generation.
